control/veda_file_discovery.py

```diff
@@ -125,162 +127,199 @@ def about_video_validate(self, meta, key):
+            2 - Download video to node working directory from S3.
+            3 - Check if this video has valid 3rd Party transcript provider along with the preferences.
+            4 - Set up an ingest instance and insert video to ingestion phase.
+            5 - On completing ingestion, mark the video file as processed.
 
+            Note:
+                Failure at any discovery point will cause video file to be marked as rejected.
+        """
+        client_title = video_s3_key.get_metadata('client_video_id')
+        course_hex = video_s3_key.get_metadata('course_video_upload_token')
+        course_id = video_s3_key.get_metadata('course_key')
+        transcript_preferences = video_s3_key.get_metadata('transcript_preferences')
+        filename = os.path.basename(video_s3_key.name)
+
+        # Try getting course based on the S3 metadata set on the video file.
+        course = self.get_course(course_id, course_hex=course_hex)
+        if course:
+            # Download video file from S3 into node working directory.
+            file_extension = os.path.splitext(client_title)[1][1:]
+            file_downloaded = self.download_video_to_working_directory(video_s3_key, filename, file_extension)
+            if not file_downloaded:
+                # S3 Bucket ingest failed, move the file rejected directory.
+                self.move(video_s3_key, destination_dir=self.auth_dict['edx_s3_rejected_prefix'])
+                return
+
+            # Prepare to ingest.
+            video_metadata = dict(
+                s3_filename=filename,
+                client_title=client_title,
+                file_extension=file_extension,
+                platform_course_url=course_id,
+            )
+            # Check if this video also having valid 3rd party transcription preferences.
+            transcript_preferences = self.parse_transcript_preferences(course_id, transcript_preferences)
+            if transcript_preferences is not None:
+                video_metadata.update({
+                    'process_transcription': True,
+                    'provider': transcript_preferences.get('provider'),
+                    'three_play_turnaround': transcript_preferences.get('three_play_turnaround'),
+                    'cielo24_turnaround': transcript_preferences.get('cielo24_turnaround'),
+                    'cielo24_fidelity': transcript_preferences.get('cielo24_fidelity'),
+                    'preferred_languages': transcript_preferences.get('preferred_languages'),
+                    'source_language': transcript_preferences.get('video_source_language'),
+                })
+
+            ingest = VedaIngest(
+                course_object=course,
+                video_proto=VideoProto(**video_metadata),
+                node_work_directory=self.node_work_directory
+            )
+            ingest.insert()
 
-if __name__ == '__main__':
-    sys.exit(main())
+            if ingest.complete:
```

Comment:
What if ingest is not completed? What we need to do then?

Reply:
just return, as it was doing before